Code/src/main/java/nl/utwente/viskell/ui/components/BlockContainer.java

```diff
@@ -15,4 +17,25 @@
      */
     public void handleConnectionChanges(boolean finalPhase);
 
+    /** Gets the value of the property boundsInLocal. */
+    public Bounds getBoundsInLocal();
+    
+    /** Transforms a point from the local coordinate space of this Node into the coordinate space of its scene. */
+    public Bounds localToScene(Bounds bounds);
+    
+    /** Attach a block to this container */
+    public void attachBlock(Block block);
+    
+    /** Detach a block from this container */
+    public boolean detachBlock(Block block);
+    
+    /** Detach all blocks from this container */
+    void detachAllBlocks();
+    
+    /** Check whether this container contains the specified block */
+    public boolean containsBlock(Block block);
```

Comment:
It's both. Definitely the former, strictly not the latter until containers actually contain nodes instead of having nodes hovering above them.